### PR TITLE
Card Markdown Editor

### PR DESCRIPTION
**Pull request by hsduk**
Adds a markdown editor for cards using EpicEditor

(Also includes changes to create child boards and generate cards from json url)

### CODE REVIEW

File: client/components/boards/boardBody.js [hunk 1]
```diff
@@ -186,14 +209,40 @@ BlazeComponent.extendComponent({
     return [{
       submit(evt) {
         evt.preventDefault();
+<<<<<<< HEAD
```
Comment (mquandalle) [inline client/components/boards/boardBody.js]: :zap:

**Comment by hsduk** [comment]
Epic.create('mdeditorcontainer', opts);

eslint returns:  error  "Epic" is not defined  no-undef
I just used the example from https://github.com/d4nyll/epic
Any ideas? Thx!

**Comment by mquandalle** [comment]
Yes, you should add it to the list of authorized globals, https://github.com/wekan/wekan/blob/31b60d82fcae64a844805a2a76a0af25fb9c16c2/.eslintrc#L83

File: client/components/cards/cardDetails.jade [hunk 2]
```diff
@@ -10,9 +10,13 @@ template(name="cardDetails")
...
         if currentUser.isBoardMember
           a.fa.fa-ellipsis-v.card-details-menu.js-open-card-details-menu
+        if linkedBoard
+          a.fa.fa-clone.card-details-menu.js-open-board(title="{{_ 'linked-board'}}")
+        if cardMarkdown 
+          a.fa.fa-file-text-o.card-details-menu.js-edit-markdown(title="{{_ 'linked-markdown'}}")
         h2.card-details-title.js-card-title(
           class="{{#if currentUser.isBoardMember}}js-open-inlined-form is-editable{{/if}}")
-            = title
+            = title
```
Comment (mquandalle) [inline client/components/cards/cardDetails.jade]: Please remove trailing spaces (you should configure our editor to do so).

File: client/components/cards/cardDetails.jade [hunk 3]
```diff
@@ -120,3 +137,23 @@ template(name="cardDeletePopup")
   unless archived
    p {{_ "card-delete-suggest-archive"}}
   button.js-confirm.negate.full(type="submit") {{_ 'delete'}}
+
+template(name="createLinkedBoardPopup")
```
Comment (mquandalle) [inline client/components/cards/cardDetails.jade]: Is that an unrelated feature?

File: .meteor/packages [hunk 4]
```diff
@@ -74,3 +74,7 @@ perak:markdown
 seriousm:emoji-continued
 templates:tabs
 verron:autosize
+
+http
+ejson
+d4nyll:epic
```
Comment (mquandalle) [inline .meteor/packages]: You should move `http` and `ejson` to the `Utilities` and `epic` to the UI components. Packages are sorted alphabetically — with official packages first.

File: client/components/boards/boardBody.js [hunk 5]
```diff
@@ -186,14 +204,34 @@ BlazeComponent.extendComponent({
     return [{
       submit(evt) {
         evt.preventDefault();
-        const title = this.find('.list-name-input').value.trim();
-        if (title) {
-          Lists.insert({
-            title,
+        const title = this.find('.list-name-input');
+        if ($.trim(title.value)) {
```
Comment (mquandalle) [inline client/components/boards/boardBody.js]: We now use `title.value.trim()` instead of jquery utility function.

**Comment by mquandalle** [comment]
I’ve put some inline comments about stylistic issues but anyway I won’t merge this issue as is. The feature is completely bugged (at least on Firefox), and even if it worked I’m not sure about the UX.

**Comment by hsduk** [comment]
Thx - I guess this will take time to do right!

I have created a discussion at http://discuss.wekan.io/t/wekan-plugin-system/78 with some more suggestions

I've been working on adding a Key/Value addin since as this will be more useful for me than an md editor as can be used to add keys and values to cards such as:

DueDate
CompletedDate
CardStatus
Predecessors

i thought about using redis but will be overkill
plan to extend this to store key/values at board and list level too which are inherited by the card - and able to be overridden
plan to facilitate an api so external systems can read/update kv's too

nonetheless, i will send a better pr soon

**Comment by hsduk** [comment]
closing this for now as working on a simpler way to add functionality

**Pull request closed without merging**